plugins/scaffolder-backend/src/scaffolder/stages/publish/helpers.ts

```diff
@@ -67,3 +68,61 @@ export async function initRepoAndPush({
     remote: 'origin',
   });
 }
+
+type BranchProtectionOptions = {
+  client: Octokit;
+  owner: string;
+  repoName: string;
+  isRetry?: boolean;
+};
+
+export const enableBranchProtectionOnDefaultRepoBranch = async ({
+  repoName,
+  client,
+  owner,
+  isRetry = false,
+}: BranchProtectionOptions): Promise<void> => {
+  const { data: repo } = await client.repos.get({
+    owner,
+    repo: repoName,
+  });
+
+  try {
+    await client.repos.updateBranchProtection({
+      headers: {
+        Accept:
+          /**
+           * 👇 we need this header because allowing a custom
+           * reviewer count on branch protection is a preview
+           * feature.
+           *
+           * More here: https://docs.github.com/en/rest/overview/api-previews#require-multiple-approving-reviews
+           */
+          'application/vnd.github.luke-cage-preview+json',
```

Comment:
Setting [mediaType.previews](https://octokit.github.io/rest.js/v18#previews) is a slightly nicer syntax with Octokit. You can also set it on the client as a whole (as we do with `nebula-preview`), but probably saner to leave it alongside the call that needs it specifically here.

Reply:
This is super helpful! I did not know Octokit had that as part of its API. Thanks!